Move xfn out-of-tree

phisco opened the pull request:
<!--
Thank you for helping to improve Crossplane!

Please read through https://git.io/fj2m9 if this is your first time opening a
Crossplane pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes

<!--
Briefly describe what this pull request does, and how it is covered by tests.
Be proactive - direct your reviewers' attention to anything that needs special
consideration.

You MUST either [x] check or ~strikethrough~ every item in the checklist below.

We love pull requests that fix an open issue. If yours does, use the below line
to indicate which issue it fixes, for example "Fixes #500".
-->

Partially address https://github.com/crossplane/crossplane/issues/4432, XFN will be moved to a dedicated repository and updated to implement the new xfn interface defined in https://github.com/crossplane/crossplane/pull/4431.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit **and** E2E tests for my change.
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR if necessary.~

[contribution process]: https://git.io/fj2m9

phisco committed:
feat: remove xfn

Signed-off-by: Philippe Scorsolini <p.scorsolini@gmail.com>

negz commented on cmd/crossplane/core/certs.go:
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 */
 
-package oci
+package core
```
What still uses the functionality in this file? I couldn't immediately find anything.

phisco replied:
It's used here in the same package: https://github.com/crossplane/crossplane/blob/e312b6e92b85d781718f20a29547f69ad6e7558a/cmd/crossplane/core/core.go#L237

negz commented on go.mod:
```diff
@@ -30,15 +26,19 @@ require (
+	github.com/google/uuid v1.3.0 // indirect
+	golang.org/x/sys v0.10.0 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20230525234030-28d5490b6b19 // indirect
+)
```
Does it make sense for these to all be in their own require block? Should they be merged with the following group?

negz reviewed (approved):
This LGTM! I'd prefer to merge sooner rather than later, since I'm about to start working on the PTFComposer. As long as the build passes we can remove anything we missed later.

negz commented:
@phisco Looks like this now needs a rebase.

negz commented:
Note: Code has moved to https://github.com/crossplane/function-runtime-oci/pull/1

phisco commented:
Ops, I guess it's for the E2Es, I'm not at my laptop now, if you want feel free to do that, otherwise I'll do it tomorrow morning 🙏

negz committed:
Remove Composition Function test images

We want to test these as part of https://github.com/crossplane/function-runtime-oci

Signed-off-by: Nic Cope <nicc@rk0n.org>
